sql/core/src/main/scala/org/apache/spark/sql/execution/python/UserDefinedPythonFunction.scala

```diff
@@ -106,7 +106,7 @@ case class UserDefinedPythonTableFunction(
     this(name, func, None, pythonEvalType, udfDeterministic)
   }
 
-  def builder(exprs: Seq[Expression]): LogicalPlan = {
+  def builder(exprs: Seq[Expression], parser: ParserInterface): LogicalPlan = {
```

Comment:
This should be `parser: => ParserInterface` to be a by-name argument. https://github.com/apache/spark/pull/43946#discussion_r1408346486

Reply:
This is done.